### PR TITLE
pythonPackages.nimfa: init at 1.3.1

### PR DESCRIPTION
**Pull request by ashgillman**
###### Motivation for this change
New Python package for nonnegative matrix factorization (maths and science).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/nimfa/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+, buildPythonPackage
+, fetchPypi
+, numpy
+, scipy
+, matplotlib
+, pytest
+}:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/nimfa/default.nix]: no `name`

File: pkgs/development/python-modules/nimfa/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "nimfa";
+  version = "1.3.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "05d0m5n96bg6wj94r7m1har48f93797gk5v9s62zdv7x83a6n6j5";
+  };
+
+  propagatedBuildInputs = [ numpy scipy ];
+  buildInputs = [ matplotlib pytest ];
```
Comment (FRidh) [inline pkgs/development/python-modules/nimfa/default.nix]: checkInputs

File: pkgs/development/python-modules/nimfa/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  doCheck = false; # errors
```
Comment (FRidh) [inline pkgs/development/python-modules/nimfa/default.nix]: What kind of errors? Are these upstream issues? If so, did you report them?

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/nimfa/default.nix (hunk 1 above)
Comment (ashgillman) [reply pkgs/development/python-modules/nimfa/default.nix]: Python packages shouldn't have a `name`?

Comment (dotlambda) [reply pkgs/development/python-modules/nimfa/default.nix]: `buildPythonPackage` will automatically set the name according to `pname` and `version`.

**Comment by ashgillman** [comment]
I believe all changes have been made

**Pull request merged**